.github/workflows/merge-head.yml

```diff
@@ -0,0 +1,37 @@
+name: Merge to Head
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [ synchronize ]
+
+permissions:
+  actions: read
+  checks: none
+  contents: write
+  deployments: none
+  issues: read
+  packages: none
+  pull-requests: write
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout current branch
+        uses: actions/checkout@v2
+        with:
+          lfs: false
+          submodules: recursive
+          
+      - name: Merge ${{ github.base_ref }} -> ${{ github.head_ref }}
+        uses: devmasx/merge-branch@master
+        with:
+          type: now
+          from_branch: ${{ github.base_ref }}
+          target_branch: ${{ github.head_ref }}
+          github_token: ${{ github.token }}
+
```

Comment:
I actually don't think this is a good idea to add or at least to execute automatically. I'm pretty sure this will cause some trouble, when people are pushing code to branches and github automatically adds merge commits.
If we want to have branches up to date with latest 4.x-dev we can easily activate that in the options to only allow merges from branches that are up to date, github than automatically provides a button to update the branch. For example see `https://github.com/matomo-org/device-detector/pull/6296`